src/syzygy/tbcore.cpp

```diff
@@ -288,83 +287,62 @@ void Tablebases::init(const std::string& path)
     if (path.empty() || path == "<empty>")
         return;
 
-    // Tokenize path into paths[] using SEP_CHAR delimiter
-    std::string s(path);
-    size_t pos = 0;
+    // Argument path is set to the directory or directories where the .rtbw and
+    // .rtbz files can be found. Multiple directories are separated by ";" on
+    // Windows and by ":" on Unix-based operating systems.
+    //
+    // Example:
+    // C:\tb\wdl345;C:\tb\wdl6;D:\tb\dtz345;D:\tb\dtz6
 
-    while ((pos = s.find(SEP_CHAR)) != std::string::npos) {
-        paths.push_back(s.substr(0, pos));
-        s.erase(0, pos + 1);
-    }
+    // Tokenize path into paths[] using SEP_CHAR delimiter
+    std::stringstream ss(path);
+    std::string token;
 
-    paths.push_back(s);
+    while (std::getline(ss, token, SEP_CHAR))
+        paths.push_back(token);
 
     TBnum_piece = TBnum_pawn = 0;
     MaxCardinality = 0;
 
-    for (i = 0; i < (1 << TBHASHBITS); i++)
-        for (j = 0; j < HSHMAX; j++) {
-            TB_hash[i][j].key = 0ULL;
-            TB_hash[i][j].ptr = NULL;
+    for (int i = 0; i < (1 << TBHASHBITS); i++)
+        for (int j = 0; j < HSHMAX; j++) {
+            TB_hash[i][j].key = 0;
+            TB_hash[i][j].ptr = nullptr;
         }
 
-    for (i = 0; i < DTZ_ENTRIES; i++)
-        DTZ_table[i].entry = NULL;
+    for (int i = 0; i < DTZ_ENTRIES; i++)
+        DTZ_table[i].entry = nullptr;
 
-    for (i = 1; i < 6; i++) {
-        sprintf(str, "K%cvK", pchr_rev[i]);
-        init_tb(str);
-    }
+    const std::string K("K");
```

Comment:
On Sun, Apr 17, 2016 at 2:14 PM, lucasart <notifications@github.com> wrote:

> In src/syzygy/tbcore.cpp
> <https://github.com/official-stockfish/Stockfish/pull/637#discussion_r59983088>
> :
>
> >
> > -    for (i = 1; i < 6; i++) {
> > -        sprintf(str, "K%cvK", pchr_rev[i]);
> > -        init_tb(str);
> > -    }
> > +    const std::string K("K");
>
> This is weird. Why not remove this, and replace K, by "K" ?
>
>
Because it does not work, you need a std::string if you want to place an
operator+() at his right side.